_posts/2021-08-23-using-lra.adoc

```diff
@@ -0,0 +1,490 @@
+
+IMPORTANT: The extension provides an implementation of the https://download.eclipse.org/microprofile/microprofile-lra-1.0-M1/microprofile-lra-spec.html[MicroProfile LRA specification] and its associated
+ https://download.eclipse.org/microprofile/microprofile-lra-1.0/apidocs/index.html?org/eclipse/microprofile/lra/annotation/ws/rs/LRA.html[annotations].
+
+== LRA coordinators
+
+The `narayana-lra-extension` requires the presence of a running coordinator in the environment. Coordinators can be obtained from `https://hub.docker.com/r/jbosstm/lra-coordinator`
+or you can build your own coordinator using a maven pom that includes the appropriate dependencies. For the purpose of this blog we'll show how to create from scratch using the `quarkus-maven-plugin`.
+There is some extra information about configuring the coordinator in one of the https://jbossts.blogspot.com/2021/07/narayana-lra-update.html[narayana blogs].
+
+Since the coordinator is just a JAX-RS resource we can build one using quarkus, adding the `resteasy-jackson` and `rest-client` extensions:
+
+[source,bash]
+----
+$ mvn io.quarkus:quarkus-maven-plugin:2.2.0.Final:create \
+      -DprojectGroupId=org.acme \
+      -DprojectArtifactId=narayana-lra-coordinator \
+      -Dextensions="resteasy-jackson,rest-client"
+$ cd narayana-lra-coordinator/
+$ rm -rf src
+----
+
+Update the pom.xml file to add a dependency on the narayana coordinator implementation:
+
+[source,xml]
+----
+    <dependency>
+      <groupId>org.jboss.narayana.rts</groupId>
+      <artifactId>lra-coordinator-jar</artifactId>
+      <version>5.12.0.Final</version>
+    </dependency>
+----
+
+Now build it and run it in the background:
+
+[source,bash]
+----
+$ ./mvnw clean package
+$ java -Dquarkus.http.port=50000 -jar target/quarkus-app/quarkus-run.jar &
+----
+
+Here we are running the coordinator on the default port used by the `narayana-lra` quarkus extension, namely `50000`.
+You can verify that the coordinator is running okay by listing the current LRAs:
+
+[source,bash]
+----
+$ curl http://localhost:50000/lra-coordinator
+[]
+----
+
+This snippet shows the request returning an empty json array.
+
+We shall leave the coordinator running while we develop and test services that use LRAs.
+Towards the end of article we will show how to embed coordinators with services
+(NOTE: you cannot use this approach to run coordinators in native mode, a future extension will be provided to support this requirement).
+
+== Non native mode
```

Comment:
I was wanting to draw out the distinction (between native and JVM) but I guess I can reword it if you think it's confusing.

Reply:
I was just thinking that the Quarkus usually use the term "JVM mode". My point was to be consistent in the usual naming around Quarkus. I do understand both. If the current term shows better the point it can be used as it is.

Reply:
I have gone with your suggestion.